tracer.go

```diff
@@ -26,22 +26,7 @@ type Tracer interface {
 	// Like `StartTrace`, but the returned `Span` is made a child of `parent`.
 	//
 	// The `parent` parameter can either be a `context.Context` or an
-	// `opentracing.TraceContext`. In the former case, the implementation
-	// attempts to extract an `opentracing.Span` using `SpanFromGoContext()`.
+	// `opentracing.Span`. In the former case, the implementation attempts to
+	// extract an `opentracing.Span` using `SpanFromContext()`.
 	JoinTrace(operationName string, parent interface{}) Span
```

Comment:
I don't like the startChild function being on the Span. It mixes Span creation with behavior. I think it is better to collect all creation functions into the Tracer.